### PR TITLE
Rewrote js/mage/adminhtml/variables.js without prototypejs

### PR DESCRIPTION
**Pull request by fballiano**
This can be tested going to "CMS -> static blocks", opening a block and they using the "insert variable" button, this js handles the opening of this overlay:

Screenshot 2024-01-26 alle 18 45 44

### CODE REVIEW

File: js/mage/adminhtml/variables.js [hunk 1]
```diff
@@ -8,7 +8,7 @@
  * @category    Mage
  * @package     Mage_Adminhtml
  * @copyright   Copyright (c) 2006-2020 Magento, Inc. (https://www.magento.com)
- * @copyright   Copyright (c) 2022 The OpenMage Contributors (https://www.openmage.org)
+ * @copyright   Copyright (c) 2022-2024 The OpenMage Contributors (https://www.openmage.org)
```
Comment (pquerner) [inline js/mage/adminhtml/variables.js:11]: I still dont like that. A more common thing would be "see COPYRIGHT.md" file?! And only update once and not in 99999 files? :D 
</rant over>

Comment (fballiano) [reply js/mage/adminhtml/variables.js:11]: that can be discussed but it's outside the scope of this PR.

Comment (sreichel) [reply js/mage/adminhtml/variables.js:11]: @pquerner +1

At he one hand i'm okay with it, at the other it causes changes that can be avoided (for a cleaner history).

File: js/mage/adminhtml/variables.js [hunk 2]
```diff
@@ -20,27 +20,28 @@ var Variables = {
     overlayShowEffectOptions: null,
     overlayHideEffectOptions: null,
     insertFunction: 'Variables.insertVariable',
-    init: function(textareaElementId, insertFunction) {
-        if ($(textareaElementId)) {
+
+    init: function (textareaElementId, insertFunction) {
```
Comment (kiatng) [inline js/mage/adminhtml/variables.js:24]: I am not used to the space after `function`. Is this a thing now?

Comment (fballiano) [reply js/mage/adminhtml/variables.js]: done ;-)

**Review by kiatng — APPROVED** [review]

**Review by empiricompany — APPROVED** [review]

**Review by ADDISON74 — APPROVED** [review]

**Pull request merged**